### PR TITLE
🐛 bug(Source facebook-marketing): remove unused discriminator from spec

### PR DESCRIPTION
**Pull request by aldogonzalez8**
## What
 It appears that the registry was updated, and Facebook Marketing’s spec now contains a credentials block that references definitions that are not found anywhere.
```
 "credentials": {
              "title": "Authentication",
              "description": "Credentials for connecting to the Facebook Marketing API",
              "type": "object",
              "discriminator": {
                "propertyName": "auth_type",
                "mapping": {
                  "Client": "#/definitions/OAuthCredentials",
                  "Service": "#/definitions/ServiceAccountCredentials"
                }
              },
```
That’s causing issues with generating the terraform provider. 

[Here is the conversation](https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1720657497281509) related if you want to make a deep dive.

## How
Let's remove unused discriminator

## Review guide

1. `integration_tests/spec.json`: discrimintor being removed

## User Impact
Terraform provider can be generated now

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by aldogonzalez8** [comment]
@cmm-airbyte @bazarnov 
The live test tool [was able to run ](https://github.com/airbytehq/airbyte/actions/runs/9897148316)for this PR, I was trying to do more tests locally, but I'm having some problems with my local internet, is there any special connection id you want to run these changes or do you feel good with this run.

image


image

**Review by JonsSpaghetti — COMMENTED** [review]
I'm not sure I'm the right person to review this, but this seems like it would indeed get rid of the thing that's causing my issue.

I think the main thing I want to make sure of as someone who isn't super familiar with which files end up affecting the registry is - is the spec.json file in integration_tests the file that we care about? The fact that it's in integration_tests suggests no, but again I know relatively little about this area.

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json [hunk 1]
```diff
@@ -30,13 +30,6 @@
         "title": "Authentication",
         "description": "Credentials for connecting to the Facebook Marketing API",
         "type": "object",
-        "discriminator": {
```
Comment (girarda) [inline airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json:33]: what is the discriminator used for?

**Comment by JonsSpaghetti** [comment]
@aldogonzalez8 looks like there's a version increment check failing FYI

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py [hunk 2]
```diff
@@ -202,7 +200,6 @@ class Config:
     credentials: Optional[Union[OAuthCredentials, ServiceAccountCredentials]] = Field(
```
Comment (cmm-airbyte) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:200]: without the discriminator field you should remove "Optional" here as well.

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json (hunk 1 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json:33]: It is used to distinguish between different schema definitions based on a property value. In this case I think this is a little overloaded as we also are using also OneOf to have one of either schemas

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py (hunk 2 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:200]: Actually, I'm starting to feel less comfortable with this approach as I'm seeing these objects are gone from required as I remove the discriminator and that feels odd 

image

Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:200]: Actually ignore previous comment, let me do more testing on this.

Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:200]: Actually this comment was very useful, thanks @cmm-airbyte

**Comment by aldogonzalez8** [comment]
@bazarnov @cmm-airbyte, with current changes, this is the result diff of the master vs. fix branch (_poetry run source-facebook-marketing spec_).

 Can you confirm that this looks good to you?

1.- Unused discriminator in credentials is gone (unused because the #defined paths won't exist).
2.- Credentials fields was added to required

The spec.json was updated to sync with spec command results.

image

image

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py [hunk 3]
```diff
@@ -199,10 +199,9 @@ class Config:
         airbyte_secret=True,
     )
 
-    credentials: Optional[Union[OAuthCredentials, ServiceAccountCredentials]] = Field(
+    credentials: Union[OAuthCredentials, ServiceAccountCredentials] = Field(
         title="Authentication",
         description="Credentials for connecting to the Facebook Marketing API",
-        discriminator="auth_type",
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py:205]: @cmm-airbyte  I'm just curious about why this used to be optional, it seems credentials are something we always wanted to be required(?).  You can look at the spec command results I left in the conversation tab. cc @bazarnov

**Comment by aldogonzalez8** [comment]
Service authentication and Oauth worked fine, I set [this workspace](https://cloud.airbyte.com/workspaces/e51d7269-6c34-4539-b3bd-376d19371ce0/connections) for my testing if you wish to take a look

image

**Review by bazarnov — APPROVED** [review]

**Comment by aldogonzalez8** [comment]
I re-run live tests after the last change with a couple of streams:

ads_insights_region
image

ads_insights_dma
image

**Pull request merged**